src/bin/pg_autoctl/cli_common.c

```diff
@@ -1292,12 +1292,21 @@ cli_pg_autoctl_reload(const char *pidfile)
 
 	if (read_pidfile(pidfile, &pid))
 	{
+		if (pid <= 0)
+		{
+			log_error("Failed to reload pg_autoctl: "
+					  "pid file \"%s\" contains negative-or-zero pid %d",
+					  pidfile, pid);
+			return false;
+		}
+
 		if (kill(pid, SIGHUP) != 0)
 		{
 			log_error("Failed to send SIGHUP to the pg_autoctl's pid %d: %m",
 					  pid);
 			return false;
 		}
 	}
-	return true;
+
+	return false;
```

Comment:
This change seems incorrect, since now it always fails.

Reply:
Good catch, fixed now!